firedrake/supermeshing.py

```diff
@@ -20,15 +20,17 @@
 from pyop2.compilation import load
 from pyop2.mpi import COMM_SELF
 from pyop2.utils import get_petsc_dir
+from collections import defaultdict
 
 
 __all__ = ["assemble_mixed_mass_matrix", "intersection_finder"]
 
 
 class BlockMatrix(object):
-    def __init__(self, mat, dimension):
+    def __init__(self, mat, dimension, symmetry=None):
```

Comment:
A docstring explaining what `symmetry` actually is and what it does would be great.

Reply:
Why can't we use a KAIJ Mat? This code isn't going through PyOP2